bioptim/misc/mapping.py

```diff
@@ -191,8 +191,15 @@ def add(
             The BiMapping to copy
         """
 
-        if name in self:
-            raise ValueError("BiMapping name should be unique")
+        multiple_phase_option = False
+        if phase == -1:
+            multiple_phase_option = True
+            if name in self.options:
+                raise ValueError(f"BiMapping name should be unique inside each phase. {name} is not unique.")
+        else:
+            if len(self.options) > phase:
+                if name in self.options[phase]:
+                    raise ValueError(f"BiMapping name should be unique inside each phase. {name} is not unique.")
```

Comment:
This should be done at next level (the next `add` can perform this task more efficiently. We can have a look together for that part